src/App.tsx

```diff
@@ -1,61 +1,123 @@
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {selectedUser && isLoading && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {!isLoading && isError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {selectedUser
+                  && !isLoading
+                  && !isError
+                  && posts.length === 0
+                  && (
+                    <div
+                      className="notification is-warning"
+                      data-cy="NoPostsYet"
+                    >
+                      No posts yet
+                    </div>
+                  )}
+
+                {selectedUser
+                  && !isLoading
+                  && !isError
+                  && posts.length > 0
+                  && (
+                    <PostsList
+                      posts={posts}
+                      selectedPost={selectedPost}
+                      setSelectedPost={setSelectedPost}
+                    />
+                  )}
 
-                <PostsList />
               </div>
             </div>
           </div>
 
           <div
             data-cy="Sidebar"
-            className={classNames(
+            className={cn(
               'tile',
               'is-parent',
               'is-8-desktop',
               'Sidebar',
-              'Sidebar--open',
+              { 'Sidebar--open': selectedPost },
             )}
           >
-            <div className="tile is-child box is-success ">
-              <PostDetails />
-            </div>
+            {selectedPost && (
+              <div className="tile is-child box is-success ">
+                <PostDetails
```

Comment:
The PostDetails component is missing propTypes. Including propTypes can help catch bugs and improve the readability and maintainability of your code.

Reply:
<img width="356" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-posts/assets/25157075/85b6cda5-7212-45e1-87c3-a534e7a9efe6">

There are propTypes in PostDetails.tsx file